frontend/src/features/map_planning/components/timeline/ItemSliderPicker.tsx

```diff
@@ -0,0 +1,307 @@
+    if (isScrollingToIndex.current) {
+      isScrollingToIndex.current = false;
+      return;
+    }
+
+    const selectedItemIndex: number = identifySelectedIndexBasedOnScrollPosition();
+    setSelectedItemAndDetectIfBordersAreReached(items[selectedItemIndex].key);
+  };
+
+  const handleMouseDown = (e: MouseEvent) => {
+    const container = sliderContainerRef.current;
+
+    if (container) {
+      isDragging.current = true;
+      dragStartX.current = e.clientX;
+
+      scrollPositionOnMouseDown.current = container.scrollLeft;
+    }
+  };
+
+  const handleMouseMove = (e: MouseEvent) => {
+    if (!isDragging.current || dragStartX.current === null) return;
+
+    const container = sliderContainerRef.current;
+    if (container && scrollPositionOnMouseDown.current) {
+      const deltaX = e.clientX - dragStartX.current;
+
+      container.scrollTo({
+        behavior: 'auto',
+        left: scrollPositionOnMouseDown.current - deltaX,
+      });
+    }
+  };
+
+  const handleItemClick = (e: React.MouseEvent, index: number) => {
+    const container = sliderContainerRef.current;
+    if (container && scrollPositionOnMouseDown.current === container.scrollLeft) {
+      setSelectedItemAndNotifyParent(items[index].key);
+      scrollToIndex(index);
+    }
+  };
+
+  const handleMouseUp = () => {
+    if (
+      isDragging.current &&
+      scrollPositionOnMouseDown.current !== sliderContainerRef.current?.scrollLeft
+    ) {
+      scrollToIndex(getSelectedItemIndex());
+      onChange(selectedItemKey);
+    }
+
+    isDragging.current = false;
+    dragStartX.current = null;
+  };
+
+  const handleMouseWheel = (e: WheelEvent) => {
+    const container = sliderContainerRef.current;
+    if (container) {
+      e.preventDefault();
+
+      if (e.deltaY > 0) {
+        selectPreviousItem();
+      } else {
+        selectNextItem();
+      }
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'ArrowLeft') {
+      selectPreviousItem();
+      e.stopPropagation();
+      e.preventDefault();
+    } else if (e.key === 'ArrowRight') {
+      selectNextItem();
+      e.stopPropagation();
+      e.preventDefault();
+    }
+  };
+
+  useEffect(() => {
+    const container = sliderContainerRef.current;
+    container?.addEventListener('scroll', handleScroll);
+    container?.addEventListener('mousedown', handleMouseDown);
+    container?.addEventListener('wheel', handleMouseWheel);
+    document.addEventListener('mousemove', handleMouseMove);
+    document.addEventListener('mouseup', handleMouseUp);
+
+    return () => {
+      container?.removeEventListener('scroll', handleScroll);
+      container?.removeEventListener('mousedown', handleMouseDown);
+      container?.removeEventListener('wheel', handleMouseWheel);
+      document.removeEventListener('mousemove', handleMouseMove);
+      document.removeEventListener('mouseup', handleMouseUp);
+    };
+  });
```

Comment:
Can this also be assigned in the JSX with `onScroll` etc?
React then handles the removal of the event listeners automatically.
Also this effect is called every render.